lib/pbench/server/api/resources/query_apis/__init__.py

```diff
@@ -512,17 +532,34 @@ def post(self):
         validation/conversion where necessary. Missing "required" parameters
         and any parameter with a null value are rejected. Parameters that
         are not in the class schema are ignored but logged.
+
+        If the request does not contain the user field, it will be interpreted
+        as a public dataset query.
         """
         json_data = request.get_json(silent=True)
+        if json_data and "user" in json_data:
+            if not json_data["user"]:
+                # There is an empty user field present in the json data;
+                # therefore, this request will be interpreted as a
+                # public dataset query.  Remove it before validating the
+                # remaining json data.
+                del json_data["user"]
+            elif Auth.token_auth.current_user() is None:
+                # Only logged-in users are allowed to query non-public data.
+                abort(HTTPStatus.FORBIDDEN, message="Not Authorized")
```

Comment:
The difference between "missing" and "empty" and between defaulting to "public" vs "mine" really belongs in the validator, not out here. I'll be replacing this anyway as I make the distinction more explicit; but it occurs to me that this is "entangled" as the validator will need to know the authorization user in order to produce the proper internal translation in all cases. Therefore it might make sense to clean this up and pass the authorized user down into the validator (line 551), moving this special-case code into the username validation.

However aside from this "subtle hint" I won't push on that for this PR. :-)

Reply:
Okay yeah if its going to get replaced, I would just keep it as it is for now :)